docs/bedrock.md

```diff
@@ -0,0 +1,360 @@
+- **Mistral**: Multilingual and specialized models
+- **Stability AI**: Image generation and processing models
+
+## Prerequisites
+
+1. **AWS Account**: You need an AWS account with appropriate permissions
+2. **AWS CLI**: Install and configure AWS CLI (optional but recommended)
+3. **Model Access**: Request access to the models you want to use in AWS Bedrock console
+4. **kubectl-ai**: Version 0.0.12 or later
+
+## Setup
+
+### 1. AWS Credentials
+
+Configure your AWS credentials using one of these methods:
+
+#### Option A: AWS CLI (Recommended)
+```bash
+aws configure
+```
+
+#### Option B: Environment Variables
+```bash
+export AWS_ACCESS_KEY_ID=your_access_key
+export AWS_SECRET_ACCESS_KEY=your_secret_key
+export AWS_DEFAULT_REGION=us-west-2
+```
+
+#### Option C: AWS SSO
+```bash
+aws configure sso
+```
+
+#### Option D: IAM Role (for EC2/ECS/Lambda)
+AWS SDK will automatically use the IAM role attached to your instance.
+
+### 2. Required AWS Permissions
+
+Ensure your AWS credentials have the following permissions:
+
+```json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": [
+                "bedrock:InvokeModel",
+                "bedrock:InvokeModelWithResponseStream",
+                "bedrock:GetFoundationModel",
+                "bedrock:ListFoundationModels"
+            ],
+            "Resource": "*"
+        }
+    ]
+}
+```
+
+### 3. Request Model Access
+
+1. Go to the [AWS Bedrock Console](https://console.aws.amazon.com/bedrock)
+2. Navigate to "Model access" in the left sidebar
+3. Request access to the models you want to use
+4. Wait for approval (this can take a few minutes to hours)
+
+## Configuration
+
+### Basic Configuration
+
+Create or edit `~/.config/kubectl-ai/config.yaml`:
+
+```yaml
+llm-provider: bedrock
+model: us.anthropic.claude-sonnet-4-20250514-v1:0
+```
+
+### Advanced Configuration
+
+```yaml
+llm-provider: bedrock
+model: us.anthropic.claude-sonnet-4-20250514-v1:0
+temperature: 0.7
+max-tokens: 4000
+top-p: 0.9
+max-retries: 3
+region: us-west-2
+timeout: 30s
+```
+
+## Usage
+
+### Basic Usage
+
+```bash
+# Use default Bedrock model
+kubectl-ai --llm-provider bedrock
+
+# Use specific model
+kubectl-ai --llm-provider bedrock --model us.anthropic.claude-sonnet-4-20250514-v1:0
+
+# Use with specific region
+kubectl-ai --llm-provider bedrock --model anthropic.claude-v2:1 --region us-east-1
+```
+
+### Interactive Mode
+
+```bash
+kubectl-ai --llm-provider bedrock
+> list all pods in kube-system namespace
+> create a deployment for nginx with 3 replicas
+> help me troubleshoot why my pod is pending
+```
+
+### One-shot Commands
+
+```bash
+# Quick queries
+kubectl-ai --llm-provider bedrock --quiet "show me failing pods"
+
+# Pipe input
+echo "create a configmap from my .env file" | kubectl-ai --llm-provider bedrock
+```
+
+## Available Models
+
+### By Region
+
+The following models are available in different AWS regions:
```

Comment:
This list will keep on changing, so let's just add a link to a page (source of truth) where this is documented instead of duplicating it here.